src/ChannelServer/Network/ChannelPacketSender.cs

```diff
@@ -4106,13 +4106,113 @@ public static void ZC_UPDATED_MONSTERAPPEARANCE(Monster monster)
 		}
 
 		/// <summary>
-		/// Add a buff to the character
+		/// Display a buff on client UI
+		/// </summary>
+		/// <param name="entity"></param>
+		/// <param name="buff"></param>
+		public static void ZC_BUFF_ADD(IEntity entity, Buff buff)
+		{
+			var packet = new Packet(Op.ZC_BUFF_ADD);
+			packet.AddBuff(buff);
+
+			entity.Map.Broadcast(packet);
+		}
+
+		/// <summary>
+		/// Update a buff after a buff has been added
+		/// </summary>
+		/// <param name="entity"></param>
+		/// <param name="buff"></param>
+		public static void ZC_BUFF_UPDATE(IEntity entity, Buff buff)
+		{
+			var packet = new Packet(Op.ZC_BUFF_UPDATE);
+			packet.AddBuff(buff);
+
+			entity.Map.Broadcast(packet);
+		}
+
+		/// <summary>
+		/// Remove buff from client UI
+		/// </summary>
+		/// <param name="entity"></param>
+		/// <param name="buff"></param>
+		public static void ZC_BUFF_REMOVE(IEntity entity, Buff buff)
+		{
+			var packet = new Packet(Op.ZC_BUFF_REMOVE);
+			packet.PutInt(entity.Handle);
+			packet.PutInt((int)buff.Id);
+			packet.PutByte(0);
+			packet.PutInt(buff.Handle);
+			packet.PutByte(0);
+
+			entity.Map.Broadcast(packet);
+		}
+
+		/// <summary>
+		/// Buff list, sent when a entity spawns?
+		/// </summary>
+		/// <param name="entity"></param>
+		public static void ZC_BUFF_LIST(IEntity entity)
+		{
+			var buffs = entity.Components.Get<BuffCollection>();
+			var buffCount = buffs?.Count ?? 0;
+			var packet = new Packet(Op.ZC_BUFF_LIST);
+			packet.PutInt(entity.Handle);
+			packet.PutByte((byte)buffCount);
+			if (buffCount > 0)
```

Comment:
Please use curly braces whenever there's more than one line in the if's scope.

Reply:
Lol, I had the curly braces but removed them.